api/compute/dataset_persist.go

```diff
@@ -299,41 +352,30 @@ func splitTrainTest(sourceFile string, trainFile string, testFile string, hasHea
 			break
 		}
 	}
-
-	writerTrain.Flush()
-	writerTest.Flush()
-
-	err = util.WriteFileWithDirs(trainFile, outputTrain.Bytes(), os.ModePerm)
-	if err != nil {
-		return errors.Wrap(err, "unable to output train data")
-	}
-
-	err = util.WriteFileWithDirs(testFile, outputTest.Bytes(), os.ModePerm)
-	if err != nil {
-		return errors.Wrap(err, "unable to output test data")
-	}
-
 	return nil
 }
 
 // PersistOriginalData copies the original data and splits it into a train &
 // test subset to be used as needed.
 func PersistOriginalData(datasetName string, schemaFile string, sourceDataFolder string, tmpDataFolder string, taskType string,
-	timeseriesFieldIndex int, targetFieldIndex int) (string, string, error) {
+	timeseriesFieldIndex int, targetFieldIndex int, stratify bool) (string, string, error) {
 	// The complete data is copied into separate train & test folders.
 	// The main data is then split randomly.
 	trainFolder := path.Join(tmpDataFolder, datasetName, trainFilenamePrefix)
 	testFolder := path.Join(tmpDataFolder, datasetName, testFilenamePrefix)
 	trainSchemaFile := path.Join(trainFolder, schemaFile)
 	testSchemaFile := path.Join(testFolder, schemaFile)
 
-	// check if the data has already been split
-	log.Infof("checking folders `%s` & `%s` to see if the dataset has been previously split", trainFolder, testFolder)
-	if fileExists(trainSchemaFile) && fileExists(testSchemaFile) {
-		log.Infof("dataset '%s' already split", datasetName)
-		return trainSchemaFile, testSchemaFile, nil
-	}
+	// Disabling for now - check should also take into account whether or not stratified samplling was applied.  Would
+	// require persisting information.
+	//
+	// log.Infof("checking folders `%s` & `%s` to see if the dataset has been previously split", trainFolder, testFolder)
+	// if fileExists(trainSchemaFile) && fileExists(testSchemaFile) {
+	// 	log.Infof("dataset '%s' already split", datasetName)
+	// 	return trainSchemaFile, testSchemaFile, nil
+	// }
```

Comment:
I worry that disabling this check will cause issues from a user perspective if they are trying to compare some results across multiple solution creations.

Reply:
Added the caching back in - now takes the split params into account to ensure that re-split occurs when those parameters are modified.